src/node_v8.cc

```diff
@@ -157,6 +157,30 @@ void CachedDataVersionTag(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(result);
 }
 
+void SetHeapSnapshotNearHeapLimit(const FunctionCallbackInfo<Value>& args) {
+  CHECK(args[0]->IsNumber());
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* const isolate = args.GetIsolate();
+  int64_t limit = args[0].As<v8::Number>()->Value();
+  CHECK_GE(limit, 0);
+  // Do not set twice which will make the process crash.
+  if (limit == 0) {
+    if (env->heapsnapshot_near_heap_limit_callback_added()) {
+      env->set_heapsnapshot_near_heap_limit_callback_added(false);
+      isolate->RemoveNearHeapLimitCallback(Environment::NearHeapLimitCallback,
+                                           0);
```

Comment:
This probably shouldn't be a zero, but on the other hand, I am not quite sure what we should use as the new limit here. I'd still prefer that we don't allow resetting the limit to 0, as in general it doesn't make too much sense to disable the behavior and pretend that nothing has ever happened - if the limit has ever been hit, the application is likely to be in a bad shape and will crash soon. If the limit aren't supposed to be hit, then why bother enabling it first and then disabling it at run-time?

Reply:
Yes, i think you are right. We allow the user to set the limit by CLI or JS API only once, this makes things much simpler.